File: 21-Debugging-and-Error-Handling/72-Debugging-and-Error-Handling.py
def divide(num1, num2):
    try:
        return num1 / num2
    except TypeError:
        print("Please provide two integers or floats")
    except ZeroDivisionError:
        print("Please do not divide by zero")

print(divide(4, 2))
print(divide([], 2))
print(divide(1, 0))

File: 21-Debugging-and-Error-Handling/72-Debugging-and-Error-Handling.js
function divide(num1, num2) {
    try {
        return num1 / num2
    } 
    catch (err) {
        if (err instanceof TypeError) {
            console.log("Please provide two integers or floats")
        }
    }
}

console.log(divide(4,2))
console.log(divide('',2))  // this returns 0 ??  why?
console.log(divide([],2))  // this returns 0 ??  why?